### ISSUE TITLE
Date Add and Date Sub conversion not working for Redshift dialect

### ISSUE THREAD

**Issue by ghoshrahul** [jOOQ/jOOQ#11954]
### Expected behavior

On passing a **date_add/date_sub** function for **Redshift** dialect, it is creating incorrect query where Amazon Redshift throws an exception

### Actual behavior

Amazon Redshift is throwing an invalid operation exception

#### Result

``` 
org.jooq.exception.DataAccessException: SQL [select "users"."id" "Id", "users"."created" "Created", case when ("users"."created" + (2 * '+0-0 +0 01:00:00.000000000')) is not null then ("users"."created" + (2 * '+0-0 +0 01:00:00.000000000')) else '1970-01-01 00:00:00' end "Result" from "users" "users" -- SQL rendered with a free trial version of jOOQ 3.14.11]; [Amazon](500310) Invalid operation: invalid input syntax for integer: "+0-0 +0 01:00:00.000000000";
	at org.jooq_3.14.11.REDSHIFT.debug(Unknown Source)
	at org.jooq.impl.Tools.translate(Tools.java:2903)
	at org.jooq.impl.DefaultExecuteContext.sqlException(DefaultExecuteContext.java:757)
	at org.jooq.impl.AbstractQuery.execute(AbstractQuery.java:389)
	at org.jooq.impl.AbstractResultQuery.fetchLazy(AbstractResultQuery.java:453)
	at org.jooq.impl.AbstractResultQuery.fetchLazy(AbstractResultQuery.java:422)
	at org.jooq.impl.AbstractResultQuery.lambda$fetchStream$1(AbstractResultQuery.java:385)`
``` 

### Configuration

```java
private Settings redshiftSettings = new Settings().withParseNameCase(ParseNameCase.AS_IS).withStatementType(StatementType.STATIC_STATEMENT)
                          .withInterpreterDialect(SQLDialect.REDSHIFT).withParseDialect(SQLDialect.REDSHIFT);

DSL.using(datasource, SQLDialect.REDSHIFT, redshiftSettings)
``` 

### Steps to reproduce the problem

```java
String sqlQuery = sqlQuery = "select \"users\".\"id\" \"Id\", \"users\".\"created\" \"Created\", case when (\"users\".\"created\" + 2 * interval '1 hour') is not null then (\"users\".\"created\" + 2 * interval '1 hour') else '1970-01-01 00:00:00' end \"Result\" from \"users\" \"users\"";
ResultQuery jooqQuery = DSL.using(dslContext.configuration()).parser().parseResultQuery(sqlQuery);`
jooqQuery.getSQL() => select "users"."id" "Id", "users"."created" "Created", case when ("users"."created" + (2 * '+0-0 +0 01:00:00.000000000')) is not null then ("users"."created" + (2 * '+0-0 +0 01:00:00.000000000')) else '1970-01-01 00:00:00' end "Result" from "users" "users"
```

### Versions

- jOOQ: 3.14.11
- Java: 11
- Database (include vendor): Amazon Redshift
- OS: Ubuntu 18.04
- JDBC Driver:

This is a blocker, as our company is planning to buy the pro edition of JOOQ, for support of Amazon Redshift dialect.

**Comment by lukaseder** [comment]
Thanks a lot for your report. I'll fix this right away.

**Comment by lukaseder** [comment]
For better context, I'd love to learn more about your use-case. You're parsing Redshift SQL and execute it directly on Redshift. Why do you need jOOQ's parser for that?

**Comment by lukaseder** [comment]
Also related: https://github.com/jOOQ/jOOQ/issues/11960 Add support for Redshift interval literals

**Comment by lukaseder** [comment]
A fix for #11960 has been implemented for jOOQ 3.15.0, which also fixes this issue here. It will be available in the next nightly build from here: https://www.jooq.org/download/versions if you're licensed, or if you use an extended trial license key.

Thanks again for your report.
